src/lib/components/json/JsonInput.tsx

```diff
@@ -92,6 +96,14 @@ const JsonInput = ({ topic, text = "", setText, height }: JsonInputProps) => {
 
   const { color, response } = getResponse(jsonState);
   const isValidJson = jsonValidate(text) === null;
+
+  const showLines = useMemo(() => {
```

Comment:
not sure if useMemo is needed since if text, minLines change the component is re-rendered anyway

Reply:
As a JsonInput, I think it won't re-render. it just change only text when user keep input something.